tests/preprocessing/test_data.py

```diff
@@ -471,3 +471,109 @@ def test_inverse_transform(self):
         assert_eq_df(df, enc.inverse_transform(enc.transform(df)))
         assert_eq_df(df, enc.inverse_transform(enc.transform(df).values))
         assert_eq_df(df, enc.inverse_transform(enc.transform(df).values))
+
+
+class TestPolynomialFeatures:
+    def test_basic(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        a.fit(X)
+        b.fit(X.compute())
+        assert_estimator_equal(a._transformer, b)
+
+    def test_input_types(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        assert_estimator_equal(a.fit(df), a.fit(df.compute()))
+        assert_estimator_equal(a.fit(df), a.fit(df.compute().values))
+        assert_estimator_equal(a.fit(df.values), a.fit(df.compute().values))
+        assert_estimator_equal(a.fit(df), b.fit(df.compute()))
+        assert_estimator_equal(a.fit(df), b.fit(df.compute().values))
+
+    def test_estimator_comparison(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        a.fit_transform(X).compute()
+        b.fit_transform(X.compute())
+        assert_estimator_equal(a, b)
+
+    def test_array_transform(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        res_a = a.fit_transform(X).compute()
+        res_b = b.fit_transform(X.compute())
+        assert_eq_ar(res_a, res_b)
+
+    def test_transform_array(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        # pass numpy array to fit_transform
+        res_a1 = a.fit_transform(X.compute())
+        # pass dask array to fit_transform
+        res_a2 = a.fit_transform(X).compute()
+        res_b = b.fit_transform(X.compute())
+        assert_eq_ar(res_a1, res_b)
+        assert_eq_ar(res_a2, res_b)
+
+    def test_df_transform(self):
+        a = dpp.PolynomialFeatures(preserve_dataframe=True)
+        b = spp.PolynomialFeatures()
+
+        # pandas dataframe
+        res_pdf = a.fit_transform(df.compute()).values
```

Comment:
What's the behavior for a dask dataframe? Can you test that (if it isn't tested elsewhere)?

Reply:
Ah I see `test_daskdf_tranform` below. Any reason not to include that here? We could pametrize the test by `daskify`

```python
if not daskify:
    df = df.compute()

...

if daskify:
    assert dask.is_dask_collection(res_pdf)

assert_eq_ar(res_pdf, res_b)
```

Reply:
originally, I didn't combine it because of the dataframe comparison in `test_daskdf_transform`. But I think using `daskify` will solve the issue.